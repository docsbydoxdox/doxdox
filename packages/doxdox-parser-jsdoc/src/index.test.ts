import parse from './index';

describe('jsdoc parser', () => {
    it('parse example jsdoc headers (declaration methods)', async () => {
        await expect(
            parse(process.cwd(), './test/mocks/declaration.js')
        ).resolves.toMatchSnapshot();
    });
    it('parse example jsdoc headers (function methods)', async () => {
        await expect(
            parse(process.cwd(), './test/mocks/function.js')
        ).resolves.toMatchSnapshot();
    });
    it('parse empty file', async () => {
        await expect(
            parse(process.cwd(), './test/mocks/empty.js')
        ).resolves.toMatchSnapshot();
    });
});
