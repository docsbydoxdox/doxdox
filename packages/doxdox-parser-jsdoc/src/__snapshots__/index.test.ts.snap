// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jsdoc parser parse empty file 1`] = `
{
  "methods": [],
  "path": "./test/mocks/empty.js",
}
`;

exports[`jsdoc parser parse example jsdoc headers (declaration methods) 1`] = `
{
  "methods": [
    {
      "description": "Finds file in path.

    console.log(await findFileInPath('./', 'package.json'));
    console.log(await findFileInPath('../', 'package.json'));
    console.log(await findFileInPath('~/git/github/doxdox/', '.package.json'));",
      "fullName": "findFileInPath(input, fileName)",
      "name": "findFileInPath",
      "params": [
        {
          "description": "Directory to check for file.",
          "name": "input",
          "types": [
            "string",
          ],
        },
        {
          "description": "File name to check for.",
          "name": "fileName",
          "types": [
            "string",
          ],
        },
      ],
      "private": false,
      "returns": [
        {
          "description": "Path to package.json file.",
          "name": null,
          "types": [
            "Promise.<(string|null)>",
          ],
        },
      ],
      "slug": "test-mocks-declaration-js-findfileinpath",
    },
    {
      "description": "Get the current working directory.",
      "fullName": "getCurrentWorkingDirectory()",
      "name": "getCurrentWorkingDirectory",
      "params": [],
      "private": false,
      "returns": [
        {
          "description": "Directory path.",
          "name": null,
          "types": [
            "string",
          ],
        },
      ],
      "slug": "test-mocks-declaration-js-getcurrentworkingdirectory",
    },
    {
      "description": "Get the root directory of the package, supplied path or URL.",
      "fullName": "getRootDirPath(url)",
      "name": "getRootDirPath",
      "params": [
        {
          "description": "Optional path or URL.",
          "name": "url",
          "types": [
            "string",
          ],
        },
      ],
      "private": false,
      "returns": [
        {
          "description": "Directory path.",
          "name": null,
          "types": [
            "string",
          ],
        },
      ],
      "slug": "test-mocks-declaration-js-getrootdirpath",
    },
  ],
  "path": "./test/mocks/declaration.js",
}
`;

exports[`jsdoc parser parse example jsdoc headers (function methods) 1`] = `
{
  "methods": [
    {
      "description": "Finds file in path.

    console.log(await findFileInPath('./', 'package.json'));
    console.log(await findFileInPath('../', 'package.json'));
    console.log(await findFileInPath('~/git/github/doxdox/', '.package.json'));",
      "fullName": "findFileInPath(input, fileName)",
      "name": "findFileInPath",
      "params": [
        {
          "description": "Directory to check for file.",
          "name": "input",
          "types": [
            "string",
          ],
        },
        {
          "description": "File name to check for.",
          "name": "fileName",
          "types": [
            "string",
          ],
        },
      ],
      "private": false,
      "returns": [
        {
          "description": "Path to package.json file.",
          "name": null,
          "types": [
            "Promise.<(string|null)>",
          ],
        },
      ],
      "slug": "test-mocks-function-js-findfileinpath",
    },
    {
      "description": "Get the current working directory.",
      "fullName": "getCurrentWorkingDirectory()",
      "name": "getCurrentWorkingDirectory",
      "params": [],
      "private": false,
      "returns": [
        {
          "description": "Directory path.",
          "name": null,
          "types": [
            "string",
          ],
        },
      ],
      "slug": "test-mocks-function-js-getcurrentworkingdirectory",
    },
    {
      "description": "Get the root directory of the package, supplied path or URL.",
      "fullName": "getRootDirPath(url)",
      "name": "getRootDirPath",
      "params": [
        {
          "description": "Optional path or URL.",
          "name": "url",
          "types": [
            "string",
          ],
        },
      ],
      "private": false,
      "returns": [
        {
          "description": "Directory path.",
          "name": null,
          "types": [
            "string",
          ],
        },
      ],
      "slug": "test-mocks-function-js-getrootdirpath",
    },
  ],
  "path": "./test/mocks/function.js",
}
`;
